### PR TITLE
Shugeo resize area

### PR DESCRIPTION
**Pull request by shugeo**
## What changes were proposed in this pull request?

Implementation of image.resize_area op (as TF 1.5)

## How was this patch tested?

Proper tests were added.

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

### CODE REVIEW

File: libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Konduit K.K.
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp]: 2019-2020

File: libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                outputShape[2] = width;
+                outputShape[3] = height;
+                outputShape[4] = in[4];
+            }
+            else {
+                outputShape[1] = width;
+                outputShape[2] = height;
+                outputShape[3] = in[3];
+            }
+            ShapeUtils::updateStridesAndType(outputShape, DataType::FLOAT32, shape::order(in));
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp]: Why F32 only?

File: libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        DECLARE_SHAPE_FN(resize_area) {
+            auto shapeList = SHAPELIST(); 
+            auto in = inputShape->at(0);
+
+            Nd4jLong* outputShape;
+            auto inRank = shape::rank(in);
+            int width;
+            int height;
+            auto newImageSize = INPUT_VARIABLE(1);
+            REQUIRE_TRUE(newImageSize->lengthOf() == 2, 0, "resize_bilinear: Resize params is a pair of values, not %i.", newImageSize->lengthOf());
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/parity_ops/resize_area.cpp]: resize_area

File: libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu [hunk 4]
```diff
@@ -927,6 +928,233 @@ namespace helpers {
...
+            NDArray const* input, NDArray* output) {
+
+        T const* inputPtr = reinterpret_cast<T const*>(input->getSpecialBuffer());
+//        float* yScales;
+//        T const** yPtrs;
+        float scale = 1.f / (st.heightScale * st.widthScale);
+        auto outputPtr = reinterpret_cast<float*>(output->specialBuffer()); // output is always float. TO DO: provide another float types also with  template <typename X, typename Z> declaration
+        ImageResizerState* pSt;
+        auto err = cudaMalloc(&pSt, sizeof(ImageResizerState));
+        err = cudaMemcpy(pSt, &st, sizeof(ImageResizerState), cudaMemcpyHostToDevice);
```
Comment (raver119) [inline libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu]: please use cudaMemcpyAsync here

File: libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu [hunk 5]
```diff
@@ -927,6 +928,233 @@ namespace helpers {
...
+
+        resizeAreaKernel<T><<<128, 1, 2048, *stream>>>(pSt, cache, scale, inputPtr, input->getSpecialShapeInfo(), outputPtr,
```
Comment (raver119) [inline libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu]: Are we sure we can't make it work with more than 1 thread? :)

**Review by raver119 — CHANGES_REQUESTED** [review]
LGTM, but few minor tweaks required

File: libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu (hunk 5 above)
Comment (shugeo) [reply libnd4j/include/ops/declarable/helpers/cuda/image_resize.cu]: we can use more then 1, but not free. Now it is 4

**Pull request merged**